Add examples for each feature + minor edits (#164)
AY2223S1-CS2103T-W17-2/tp: docs/UserGuide.md

---
layout: page title: PennyWise User Guide
---

Well, hello there! Welcome to your personal budgeting assistant, PennyWise. After this guide, you will understand how to
use PennyWise, and spend every penny wisely. 🤑

## Introduction

PennyWise is a desktop application that **empowers students with the ability to make informed financial decisions**, by

> providing a **graphical analysis of their financial activities**. It provides a clean Graphical User Interface (GUI) for

Reviewer: thanks for adding this, i missed out this GUI definition

easy comprehension of expenditure and savings.
_Interested?_ Head over to [Quick Start](#quick-start) to get started. Enjoy!

<div markdown="span" class="alert alert-info">:information_source: **Info:**

> Already set PennyWise up? Jump right straight to the [features](#features) section to see what features are available!

Reviewer: ```suggestion
Already done with the set-up? Jump right straight to the [features](#features) section to see what features are available!
```

Would this sound better

Author: Yep thanks for the suggestion!

    * [Editing entries:](#editing-entries)
        * [Edits expenditure with specified fields](#edits-expenditure-with-specified-fields)
        * [Edits income with specified fields](#edits-income-with-specified-fields)
    * [Deleting entries:](#deleting-entries)
        * [Delete an expense from the expenditure list](#delete-an-expense-from-the-expenditure-list)
        * [Delete an income from the income list](#delete-an-income-from-the-income-list)
    * [Summary of entries:](#summary-of-entries)
        * [Summary of all entries in the currently shown list](#summary-of-all-entries-in-the-currently-shown-list)
        * [Summary of all entries the specified month](#summary-of-all-entries-the-specified-month)
    * [View entries by category:](#view-entries-by-category)
    * [View entries by month:](#view-entries-by-month)
    * [Clearing all entries:](#clearing-all-entries)
    * [Exiting the program:`](#exiting-the-program)
    * [Saving the data](#saving-the-data)
    * [Editing the data file](#editing-the-data-file)
* [FAQ](#faq)
* [Command summary](#command-summary)

<!-- TOC -->

## Purpose of Guide

> This guide aims to teach those new to PennyWise how to navigate and utilise the application. It also provides
> experienced or returning users a mini refresher course on how to make the most out of PennyWise.

Reviewer: ```suggestion
This guide aims to teach those new to PennyWise how to navigate and utilise the application. It also acts as 
a refresher for any experienced or returning users, allowing you to make the most out of PennyWise.
```

### Familiarising yourself with the interface

To familiarise yourself with our user interface, please refer to the following diagram and table to get comfortable with
the application.

![UserInterfaceExplanation](images/ug/UserInterfaceExplanation.png)

|      `Component`       |                                                       `Explanation`                                                        |
|:----------------------:|:--------------------------------------------------------------------------------------------------------------------------:|
|    **List
Display**    |            where you view your entry lists (Expenditure or Income) <br> - can be modified by the `view` command            |
|   **Graph
Display**    | where you view your entries in a graphical overview (Pie Chart or Line Graph) <br> - can be modified by the `view` command |
| **Command Input
Box**  |                                       where you enter your commands to use PennyWise                                       |
| **Command Output

> Box** |                                  where you see the output of your commands from PennyWise                                  |

Reviewer: Not sure why the formatting for this is adjusted, looks a bit weird when viewed in markdown

![image](https://user-images.githubusercontent.com/63844743/199515282-97ea37b2-7edd-4240-a5a2-860a65ea3e2d.png)

    </code>
</pre>

<p align="right">
    <a href="#top">Back to Top </a>
</p>

## Features

### Adding entries

> Perhaps you just bought a cup of coffee, or a paycheck comes in. Add these entries to PennyWise to keep track of them.

Reviewer 2: It's nice!

| **`[c/EDITED_CATEGORY]`**    | Updated [category](#categorising-your-expenses-and-income) that the entry belongs to                 |

#### Edits expenditure with specified fields

- Example: `edit 1 t/e d/Chicken Soup`
    - The expenditure at the 1st position will have its description edited from `Chicken Rice` to `Chicken Soup`.
- Expected: `Edited Entry: Chicken Soup; Date: 20-01-2022; Amount: 20.00; Tag: Food`

![EditCommandExpenditure](images/ug/EditCommandExpenditure.png)

#### Edits income with specified fields

- Example: `edit 2 t/i a/150.00 da/22-10-2022`
    - The income at the 2nd position will have its amount and date edited
- Expected: `Edited Entry: Monthly Allowance; Date: 02-01-2022; Amount: 150.00; Tag: Allowance`

![EditCommandIncome](images/ug/EditCommandIncome.png)

### Deleting entries

> You were going to watch the latest Kungfu Panda movie, but your friend bailed on you :(. You already logged the entry so

Reviewer 2: Should we add an user story for deleting income too?

Author: Have added @EmilyOng, do help me see it is okay :D

  user.
- The `MONTH` field is optional, if no month is specified, the application displays the summary for all entries.

### View entries by category

You've always been a foodie, but how much of your total expenditure is spent on food? You have multiple income streams,
but which helped you earn the most money? View your entries in a pie chart to find out.

Format: `view t/ENTRY_TYPE`

| Input              | Explanation                                                              |
|--------------------|--------------------------------------------------------------------------|
| **`t/ENTRY_TYPE`** | An entry type refers to either `e` for "Expenditure" or `i` for "Income" |

1. View a Pie Chart of all expenditures by categories
    - Examples: `view t/e`
    - Expected: `Show graphically all expenditure by category` and a Pie Chart on the right of the application.

   ![ViewCommandExpenditureCategory](images/ug/ViewCommandExpenditureCategory.png)

2. View a Pie Chart of all incomes by categories
    - Examples: `view t/i`
    - Expected: `Show graphically all income by category` and a Pie Chart on the right of the application

### View entries by month

> You went on a shopping spree today. Find out how today's expenditure compared to the rest of the month.

Reviewer 2: Should we talk about income too?

Author: Have added @EmilyOng, do help me see it is okay :D

Reviewer 2: Looks good!

</div>

### Exiting the program

> You're done logging your expenses for the day. Let's exit the application.

Reviewer 2: ```suggestion
You're done logging your expenses and income for the day. Let's exit the application.
```

You went on a shopping spree today. Find out how today's expenditure compared to the rest of the month.

> You were tipped for your exceptional service as a part-time server today. Find out how today's income weighs up to the

Reviewer 2: ```suggestion
Or, were you tipped for your exceptional service as a part-time server today? Find out how today's income weighs up to the
```

Not sure if the sentence started a bit abruptly from the previous line

| **
Delete**          | `del INDEX_OF_ENTRY t/ENTRY_TYPE` <br> e.g. `del 2 t/e`                                                                                                 |
| **
Edit**            | `edit INDEX_OF_ENTRY t/ENTRY_TYPE [d/EDITED_DESCRIPTION a/EDITED_AMOUNT da/EDITED_DATE c/EDITED_CATEGORY]`<br> e.g. `edit 2 t/i a/150.00 da/22-10-2022` |
| **
Summary**         | `summary [mo/MONTH]`<br> e.g. `summary mo/2022-09`                                                                                                      |
| **View (
Category)** | `view t/ENTRY_TYPE` <br> e.g. `view t/e`                                                                                                                |
| **View (
Month)**    | `view t/ENTRY_TYPE mo/MONTH` <br> e.g. `view t/i mo/2022-10`                                                                                            |
| **
Clear**           | `clear`                                                                                                                                                 |
| **

> Exit**            | `exit`                                                                                                                                                  |

Reviewer: Same issue with the table here
